### PR TITLE
Environment initial modular setup and build

### PR DESCRIPTION
**Pull request by sciecode**
### Environment Changes #3 
Complete modular environment re-structure, includes rollupjs build scripts and package dependencies.
Rollup using `esm` format for bundler.

Modular hierarchy:

    - Loader.js
    | - IFFParser.js
    | | - LWO2Parser.js
    | | - LWO3Parser.js

Currrently, only `parseBlock` function was version modularized. I intend on thinning `IFFParser` as much as possible and only keep, in that module, the functions that are used in both versions.

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * LWO2 format specification:
+ * 	http://static.lightwave3d.com/sdk/2018/html/filefmts/lwo2.html
+ *
+ * Development and Test repository:
+ *	https://github.com/threejs/lwoloader-test-models
+ *
+ **/
+ `
+
```
Comment (looeee) [inline rollup.config.js]: Let's add the version to the header: 

Version: 1.0.0

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+export default {
+	input: 'src/Loader.js',
+	output: [
+		{
+			format: 'esm',
```
Comment (looeee) [inline rollup.config.js]: We should output in at least `iife` and `esm` formats for now. 

The iife can be called `LWOLoader.js` and the esm can be called `LWOLoader.module.js`. 

We can revisit the names later, but that seems to be the consensus on the main repo for now so let's stick to it.

Comment (looeee) [reply rollup.config.js]: Oh, actually I'm wrong. It should be:

* esm: `build/jsm/LWOLoader.js`
* iife: `build/js/LWOLoader.js`

**Comment by looeee** [comment]
OK, this is looking really good, great job! 

One thing that needs to be considered - in the esm bundle, you're still using the global THREE namespace. That's unlikely to be how the module will be used though. Instead, we'll need to import the bits that we need at the top of the file. 

Take a look at the way it's being handled in the main repo:

[ESM GLTFLoader](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/loaders/GLTFLoader.js)

[IIFE GLTFLoader](https://github.com/mrdoob/three.js/blob/dev/examples/js/loaders/GLTFLoader.js)

The ESM version is auto generated from the IIFE version using this script: 
https://github.com/mrdoob/three.js/blob/dev/utils/modularize.js

That might be the easiest thing to do here for now too. That would mean that we just create the IIFE version in the rollup script, then use the modularize script to make the ESM version (you could work that into the build process too I guess).

**Comment by sciecode** [comment]
I've removed global THREE namespace from Loader and included it the bundle config. Rollup does some checking to see if namespace is valid.

    this.THREE = this.THREE || {};
    this.THREE.LWOLoader = (function () {

Included version and author to the header, hope it's alright with you.

I'll leave 'esm' version for a next PR, I wanna test the modularize script beforehand.

**Comment by looeee** [comment]
Hmm, I think eventually we'll need to remove the global THREE from src/ 

Not entirely sure how that's gonna work though so let's worry about it later.

**Pull request merged**